js/Round.js

```diff
@@ -1,30 +1,46 @@
 class Round {
-  constructor(categories, dailyDoubleCount) {
+  constructor(categories, dDCount) {
     this.categories = categories;
     this.clues = [];
-    this.dailyDoubleCount = dailyDoubleCount;
-    // this.dailyDoubleLocations = [[i, j], [i2, j2]];
+    this.dDCount = dDCount;
+    this.dDLocations = [];
   }
 
   randomizeDailyDoubles() {
-    let col = Math.floor(Math.random() * 4)
-    let row = Math.floor(Math.random() * 4)
-    this.dailyDoubleLocations.push([col, row])
+    for (let i = 0; i < this.dDCount; i++) {
+      const index = Math.floor(Math.random() * this.clues.length);
+      this.dDLocations.push(index);
+    }
+    if (this.dDCount === 2 && this.dDLocations[0] === this.dDLocations[1]) {
+      while (this.dDLocations[0] === this.dDLocations[1]) {
+        const index = Math.floor(Math.random() * this.clues.length);
+        this.dDLocations[1] = index;
+      }
+    }
```

Comment:
Instead of storing an array of the indices where daily double questions exist, it might make it a bit easier to follow this logic if we did something like this instead:

* Pull a random clue from your array  `clue[someRandomNumber]`
* Add a property to that clue (something like `dailyDouble: true`)  -- e.g. `clue[someRandomNumber].dailyDouble = true`

Then in future logic you could just check if your current clue has that `dailyDouble` property and handle it accordingly.

Reply:
Better yet, given your `setDailyDoubles` code below, you could just immediately reassign your randomly selected clue to a DailyDouble. e.g.:

`clue[someRandomNumber] = new DailyDouble(clue.question, clue.answer, etc.)`